java/download-preplanned-map-area/src/main/java/com/esri/arcgisruntime/sample/downloadpreplannedmaparea/MainActivity.java

```diff
@@ -0,0 +1,371 @@
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import android.graphics.Color;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import com.esri.arcgisruntime.ArcGISRuntimeException;
+import com.esri.arcgisruntime.concurrent.Job;
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.FeatureTable;
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.GeometryEngine;
+import com.esri.arcgisruntime.layers.Layer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+import com.esri.arcgisruntime.tasks.offlinemap.DownloadPreplannedOfflineMapJob;
+import com.esri.arcgisruntime.tasks.offlinemap.DownloadPreplannedOfflineMapParameters;
+import com.esri.arcgisruntime.tasks.offlinemap.DownloadPreplannedOfflineMapResult;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapTask;
+import com.esri.arcgisruntime.tasks.offlinemap.PreplannedMapArea;
+import com.esri.arcgisruntime.tasks.offlinemap.PreplannedUpdateMode;
+
+public class MainActivity extends AppCompatActivity implements ProgressDialogFragment.OnProgressDialogDismissListener {
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+
+  private File mOfflineMapDirectory;
+
+  private ListView mPreplannedAreasListView;
+  private List<String> mPreplannedMapAreaNames;
+  private ArrayAdapter<String> mPreplannedMapAreasAdapter;
+  private ListView mDownloadedMapAreasListView;
+  private List<String> mDownloadedMapAreaNames;
+  private ArrayAdapter<String> mDownloadedMapAreasAdapter;
+  private List<ArcGISMap> mDownloadedMapAreas;
+  private Button mDownloadButton;
+
+  private PreplannedMapArea mSelectedPreplannedMapArea;
+  private List<PreplannedMapArea> mPreplannedMapAreas;
+  private DownloadPreplannedOfflineMapJob mDownloadPreplannedOfflineMapJob;
+  private MapView mMapView;
+  private GraphicsOverlay mAreasOfInterestGraphicsOverlay;
+  private OfflineMapTask mOfflineMapTask;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
```

Comment:
Can this method be split up into other methods? It's difficult to read what it's doing.